app/mender.go

```diff
@@ -682,19 +685,39 @@ func (m *Mender) InventoryRefresh() error {
 		{Name: "mender_client_version", Value: conf.VersionString()},
 	}
 
-	if idata == nil {
-		idata = make(client.InventoryData, 0, len(reqAttr))
+	if iData == nil {
+		iData = make(client.InventoryData, 0, len(reqAttr))
 	}
-	_ = idata.ReplaceAttributes(reqAttr)
+	_ = iData.ReplaceAttributes(reqAttr)
 
-	if idata == nil {
-		log.Infof("No inventory data to submit")
+	if iData == nil {
+		log.Info("No inventory data to submit")
 		return nil
 	}
 
-	err = ic.Submit(m.api.Request(m.authToken, nextServerIterator(m), reauthorize(m)), m.Config.Servers[0].ServerURL, idata)
+	// sort slice to have a reproducible hash
+	sort.Sort(client.ByName(iData))
```

Comment:
```suggestion
	sort.Slice(func(i, j) bool { return strings.Compare(iData[i], iData[j]) < 0 })
```
With this you don't have to introduce the extra `ByName` type.

Reply:
Thanks, I'll go with your suggestion.